static/js/recent_topics_ui.js

```diff
@@ -116,6 +124,35 @@ function is_table_focused() {
     return $current_focus_elem === "table";
 }
 
+function get_row_type(row) {
+    // Return "private" or "stream"
+    // list_widget.clean_redraw() calls get_current_list before
+    // the widget is returned and thus initialized.
+    // So we use CSS method for finding row type until topics_widget gets initialized.
+    if (!topics_widget) {
+        const $topic_rows = $("#recent_topics_table table tbody tr");
+        const $topic_row = $topic_rows.eq(row);
+        const type = $topic_row.find(".recent_topic_stream a").text();
+        if (type === "Private messages") {
```

Comment:
This logic will fail once this widget is internationalized. You could check a CSS class instead, but I also wonder if we could instead just have this function not be called before the `topics_widget` is initialized and delete this block?

The comment doesn't super make sense to me; how is `get_current_list` related to this function?

Reply:
I have added the `data-private` attribute in the `tr` tag which would not pose any problem with internalization.

Reply:
OK, this is improved; but I still don't understand when this function will be called before `topics_widget` is initialized; can you explain that?

Reply:
@timabbott The bug faced here was that on reload of page, using key board navigation on first few seconds after the reload returned `topics_widget.get_current_list()` undefined.

This is due to the fact that `complete_rerender` calls `revive_current_focus` as `callback_after_render` and `revive_current_focus` calls `set_table_focus` which calls `get_row_type` which uses `get_current_list`

Reply:
OK can you open an issue with a title like "Refactor recent_topics_ui.get_row_type to not explicitly depend on topics_widget having been rendered" with a clear description of that sequence (spelled out a bit more for someone not looking at the code) and reproduction steps for the bug one would have if we removed this block? 

This logic is somewhat brittle but I think we can fix it as a follow-up issue.